cryptoki/src/mechanism/mechanism_info.rs

```diff
@@ -219,6 +218,26 @@ impl MechanismInfo {
     }
 }
 
+impl std::fmt::Display for MechanismInfo {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let flags = format!("{0:#?}", self.flags);
+        let key_size_info = match self.min_key_size == 0 && self.max_key_size == 0 {
+            true => String::new(),
+            false => {
+                if self.max_key_size == 0 {
+                    format!(", min_key_size={}", self.min_key_size)
+                } else {
+                    format!(
+                        ", min_key_size={}, max_key_size={}",
+                        self.min_key_size, self.max_key_size
+                    )
+                }
+            }
+        };
+        write!(f, "{}{}", flags, key_size_info)
```

Comment:
:thinking: I think it's a bit more complex than necessary: first `flags` are always written first so instead of `let flags = format!("{0:#?}", self.flags);` just write them directly: `write!(f, "{:#?}", self.flags);` and then you can have the additional constraint and the function should be roughly:

```rust
write!(f, "{:#?}", self.flags);
if !(self.min_key_size == 0 && self.max_key_size == 0) {

  write!(f, ", min_key_size={}", self.min_key_size);

  if self.max_size != 0 {
    write!(f, ", max_key_size={}", self.max_key_size);
  }
}
```

With no additional allocations (format! and String allocates)... phew... I hope it makes sense :sweat_smile: 

Since this is a bit involved I'm wondering if a simple test would also be in order :thinking:

Reply:
updated :)